index.bs

```diff
@@ -3830,6 +3862,804 @@ opened</dfn> steps given |window|, |type| and |message|.
+     the value of the <code>columnNumber</code> field |stack trace|[0]. Otherwise
+     let |line number| and |column number| be 0.
+
+  Otherwise, let |stack trace|, |column number|, and |line number| all be null.
+
+  TODO: Chrome includes the current parser position as column number / line
+  number for parser-inserted resources.
+
+1. Return a [=map=] matching the <code>network.Initiator</code> production, with
+   the <code>type</code> field set to |type|, the <code>columnNumber</code>
+   field set to |column number| if it's not null, or omitted otherwise, the
+   <code>lineNumber</code> field set to |line number| if it's not null, or
+   omitted otherwise, the <code>stackTrace</code> field set to |stack trace| if
+   it's not null, or omitted otherwise, and the <code>request</code> field set
+   to |request id|.
+
+</div>
+
+#### The network.Request type #### {#type-network-Request}
+
+[=Remote end definition=] and [=local end definition=]
+
+<pre class="cddl local-cddl remote-cddl">
+network.Request = text;
+</pre>
+
+Each network request has an associated <dfn export>request id</dfn>, which is a
+string uniquely identifying that request. The identifier for a request resulting from a
+redirect matches that of the request that initiated it.
+
+#### The network.RequestData type #### {#type-network-RequestData}
+
+[=Remote end definition=] and [=local end definition=]
+
+<pre class="cddl local-cddl">
+network.RequestData = {
+    request: network.Request,
+    url: text,
+    method: text,
+    headers: [*network.Header],
+    cookies: [*network.Cookie],
+    headersSize: js-uint,
+    bodySize: js-uint / null,
+    timings: network.FetchTimingInfo,
+};
+</pre>
+
+The <code>network.RequestData</code> type represents an ongoing network request.
+
+<div algorithm>
+
+To <dfn>get the request data</dfn> given |request|:
+
+1. Let |request id| be request's [=request id=].
+
+1. Let |url| be the result of running the [=URL serializer=] with |request|'s
+   [=request/URL=].
+
+1. Let |method| be |request|'s [=request/method=].
+
+1. Let |body size| be 0.
+
+1. Let |body| be request's [=request/body=].
+
+1. If |body| is a [=byte sequence=], let |body size| be the length of that
+   sequence. Otherwise, if |body| is a [=/body=] then let |body size| be that
+   body's <a spec=fetch>length</a>.
+
+   TODO: Should we handle the case where the body length is not yet known?
+
+1. Let |headers size| be the size in bytes of |request|'s [=request/headers list=] when
+   serialized as mandated by [[HTTP11]].
+
+   Note: For protocols which allow header compression, this is the compressed
+   size of the headers, as sent over the network.
+
+1. Let |headers| be an empty list.
+
+1. Let |cookies| be an empty list.
+
+1. For each (|name|, |value|) in |request|'s [=request/headers list=]:
+
+  1. Append the result of [=get a header=] with |name| and |value| to |headers|.
+
+  1. If |name| is a [=byte-case-insensitive=] match for "<code>Cookie</code>" then:
```

Comment:
If it's intentional that the cookie data will be both in the header list (previous step) and in the cookie list (following steps) can you add a note about why that makes sense?

Reply:
The reason is just "so clients don't have to manually parse the Cookie header, but can share code with the (future) get_cookies command". I can obviously just write that directly in a note, but it seems a bit odd to have that level of justification.